src/settings/toml/manifest.rs

```diff
@@ -61,11 +60,10 @@ pub struct Manifest {
     pub wasm_modules: Option<HashMap<String, PathBuf>>,
     pub triggers: Option<Triggers>,
     pub durable_objects: Option<DurableObjects>,
-    #[serde(default, with = "string_empty_as_none")]
     pub usage_model: Option<UsageModel>,
     pub compatibility_date: Option<String>,
     #[serde(default)]
-    pub compatibility_flags: Vec<String>,
+    pub compatibility_flags: Option<Vec<String>>,
```

Comment:
Why did this change? Was there some bug with it before?

Reply:
if you read and write the configuration again, it would add `compatibility_flags = []`, this change makes the round trip identical

Reply:
> if you read and write the configuration again, it would add compatibility_flags = [], this change makes the round trip identical

When would that happen, though? Are you using wrangler as a library or something?

Reply:
Yes we use wrangler as a lib to generate a project and edit the wrangler.toml

Reply:
Why not add a `skip_serializing_if = "Vec::is_empty()"` annotation instead?